src/library/scala/collection/immutable/ChampHashMap.scala

```diff
@@ -570,54 +578,70 @@ private final class BitmapIndexedMapNode[K, +V](val dataMap: Int, val nodeMap: I
 
 }
 
-private final class HashCollisionMapNode[K, +V](val originalHash: Int, val hash: Int, val content: Vector[(K, V)]) extends MapNode[K, V] {
+private final class HashCollisionMapNode[K, +V](
+  val originalHash: Int,
+  val hash: Int,
+  var contentKeys: Array[Any],
```

Comment:
@retronym with Array[Any] will operations like `indexOf`/`apply`/`update` hit the AnyRef code path or the switches for all array types?

I am suspicious of the ArrayOps uplift in this hot code

Reply:
No, it won't in that case (I checked yesterday!). It would for `Array[T]` where `T <: AnyRef`.